### ISSUE TITLE
Feature request: Add support for ES Modules with esbuild.

### ISSUE THREAD

**Issue by santiperone** [aws/aws-lambda-builders#426]
### Describe your idea/feature/enhancement

Currently, when trying to use ES modules features (as described in https://aws.amazon.com/es/blogs/compute/using-node-js-es-modules-and-top-level-await-in-aws-lambda/) together with esbuild, some errors are thrown because the default output format for esbuild is .cjs, which doesn't support those.

### Proposal

Add some option to change the output format to .mjs (maybe it could even be the default value)

**Comment by mildaniel** [comment]
Hey @santiperone. I moved the issue to the Lambda Builders repository. This is our build engine that handles the relevant logic. I also opened a pull request to add support for the `--out-extension` flag which will allow you to specify `.mjs` as the file extensions by adding this snippet to your template.

```
BuildMethod: esbuild
BuildProperties:
    OutExtension: 
     - .js=.mjs
```

As for your configuration concern, it's one we've heard before and are actively working on.
